tests/test_sparse_tensors.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+import torch
+
+# needed to register custom ops
+import xformers  # noqa: F401
+from xformers.ops import masked_matmul
+from xformers.sparse import BlockSparseTensor
+
+cuda_only = pytest.mark.skipif(not torch.cuda.is_available(), reason="requires CUDA")
+_devices = ["cuda:0"] if torch.cuda.is_available() else []
+
+
+def _create_tensor(device, BLOCK=32, Z=8, C=2, H=512, W=512, dtype=torch.float32):
+    layout = torch.randint(2, (C, H // BLOCK, W // BLOCK))
+    values = torch.randn(Z, layout.sum(), BLOCK, BLOCK, device=device).to(dtype)
+
+    mask = (
+        layout[None, :, :, None, :, None]
+        .repeat(Z, 1, 1, BLOCK, 1, BLOCK)
+        .reshape(Z, C, H, W)
+    )
+
+    return BlockSparseTensor(values, layout), mask.bool()
+
+
+@pytest.mark.parametrize("device", _devices)
+def test_masked_matmul(device):
+    BLOCK = 32
+    N, C, H, W, L = 8, 2, 512, 512, 64
+    mask_block, _ = _create_tensor(device, BLOCK, N, C, H, W, dtype=torch.bool)
+    mask = mask_block.to_dense()
+
+    a = torch.randn(N, C, H, L, device=device)
+    b = torch.randn(N, C, W, L, device=device)
+
+    aa = a.clone()
+    bb = b.clone()
+
+    b = b.transpose(-2, -1)
```

Comment:
something which could happen here (not sure) is that the kernel could assume contiguous tensors, and these are not. But even if that was the case it should probably be caught

Reply:
We check in https://github.com/facebookresearch/xformers/pull/202/files#diff-cca707218d1b441069abce210ffe653d2719bd9fadc51cc90d15a33cd918a2bcR92-R93 that the tensor is contiguous, so we should be fine here